freezing/sync/data/weather.py

```diff
@@ -70,51 +72,59 @@ def sync_weather(self, clear:bool = False, limit: int = None, cache_only: bool =
                 ride_end = ride_start + timedelta(seconds=ride.elapsed_time)
 
                 # NOTE: if elapsed_time is significantly more than moving_time then we need to assume
                 # that the rider wasn't actually riding for this entire time (and maybe just grab temps closest to start of
                 # ride as opposed to averaging observations during ride.
 
-                ride_observations = hist.find_observations_within(ride_start, ride_end)
-                start_obs = hist.find_nearest_observation(ride_start)
-                end_obs = hist.find_nearest_observation(ride_end)
+                ride_observations = [d for d in hist.hourly if ride_start <= d.time <= ride_end]
+
+                start_obs = min(hist.hourly, key=lambda d:abs((d.time - ride_start).total_seconds()))
+                end_obs = min(hist.hourly, key=lambda d:abs((d.time - ride_end).total_seconds()))
+
+                if len(ride_observations) <= 2:
+                    # if we don't have many observations, bookend the list with the start/end observations without double counting
+                    ride_observations = [start_obs] + [o for o in ride_observations if o is not start_obs and o is not end_obs] + [end_obs]
 
-                def avg(l):
-                    no_nulls = [e for e in l if e is not None]
-                    if not no_nulls:
-                        return None
-                    return sum(no_nulls) / len(no_nulls) * 1.0  # to force float
+                for x in ride_observations:
+                    self.logger.debug("Observation: {0}".format(x.__dict__))
 
                 rw = orm.RideWeather()
                 rw.ride_id = ride.id
-                rw.ride_temp_start = start_obs.temp
-                rw.ride_temp_end = end_obs.temp
-                if len(ride_observations) <= 2:
-                    # if we dont' have many observations, bookend the list with the start/end observations
-                    ride_observations = [start_obs] + ride_observations + [end_obs]
+                rw.ride_temp_start = start_obs.temperature
+                rw.ride_temp_end = end_obs.temperature
+
+                rw.ride_temp_avg = mean([o.temperature for o in ride_observations])
 
-                rw.ride_temp_avg = avg([o.temp for o in ride_observations])
+                rw.ride_windchill_start = start_obs.apparent_temperature
+                rw.ride_windchill_end = end_obs.apparent_temperature
+                rw.ride_windchill_avg = mean([o.apparent_temperature for o in ride_observations])
 
-                rw.ride_windchill_start = start_obs.windchill
-                rw.ride_windchill_end = end_obs.windchill
-                rw.ride_windchill_avg = avg([o.windchill for o in ride_observations])
+                # scale the cumulative precipitation over the observation period by the fraction of time spent moving
+                scale = timedelta(hours=len(ride_observations)).total_seconds() / ride.moving_time
```

Comment:
This is a change. Formerly, if you rode five minutes in a snowstorm you would be credited with 2 to 3 times one hour's precipitation. In the same scenario now, you would get credit for 5/60 of that hour's precipitation.